### ISSUE TITLE
Using the same “Belarusian” name for "be" and "be_tarask" codes is ambiguous

### ISSUE THREAD

**Issue by yvanzo** [houseabsolute/DateTime-Locale#32]
[`DateTime::Locale::Catalog`](https://metacpan.org/release/DROLSKY/DateTime-Locale-1.35/view/lib/DateTime/Locale/Catalog.pm) is currently naming “Belarusian” both `be` and `be-tarask`, whereas it only really fits the first code `be` which stands for the official Belarusian orthography since 1933 and the second code is for the older variant of orthography which is still informally used. See [Belarusian orthography reform of 1933](https://en.wikipedia.org/wiki/Belarusian_orthography_reform_of_1933) and [Taraškievica](https://en.wikipedia.org/wiki/Tara%C5%A1kievica).
```csv
Locale code      Locale name (in English)                    Native locale name
==========================================================================================================
be               Belarusian                                  беларуская
be-tarask        Belarusian                                  беларуская
```

If that is an upstream issue of CLDR or is it due to the way `DateTime::Locale` is importing it?

**Comment by autarch** [comment]
I'm pretty sure this is a bug in the `DateTime::Locale` generator code. The `tarask` bit is a variant, so the name should be "Belarusian Taraskievica orthography". I don't think there are any other locales in use that actually _have_ a variant that aren't simply aliases, so it's not surprising the code doesn't handle this properly.

Any interest in working on a PR to fix this? Looking at the code briefly I don't see how this bug is happening. We _should_ be including the variant in the name of the locale, but it seems like maybe the variant name is an empty string or `undef` when it should be set for this locale.

**Comment by yvanzo** [comment]
Thank you for your answer. Just reporting issues at the moment.

> I’'m pretty sure this is a bug in the `DateTime::Locale` generator code […] the name should be "Belarusian Taraskievica orthography"

From the source data, it seems that “Taraskievica orthography” should be the expected variant name indeed:
https://github.com/unicode-org/cldr-json/blob/40.0.0/cldr-json/cldr-localenames-full/main/en/variants.json#L108

The only other locale with a variant listed in  [`DateTime::Locale::Catalog`](https://metacpan.org/release/DROLSKY/DateTime-Locale-1.35/view/lib/DateTime/Locale/Catalog.pm) is `ca-ES-valencia` and is correctly named “Catalan Spain Valencian”.

The main difference here seems to be that there is a space in “Taraskievica orthography” variant name, not in “Valencian” variant name. Might it be that the generator code is not handling space correctly?

**Comment by autarch** [comment]
I don't think the space is the issue. My _guess_ is that it's because `ca-ES-valencia` also includes a territory.

**Comment by autarch** [comment]
Fixed in v1.36.

**Comment by yvanzo** [comment]
Thank you!
